paimon-core/src/main/java/org/apache/paimon/mergetree/compact/PartialUpdateMergeFunction.java

```diff
@@ -100,22 +102,44 @@ protected PartialUpdateMergeFunction(
             boolean[] nullables) {
         this.getters = getters;
         this.ignoreDelete = ignoreDelete;
-        this.fieldSeqComparators = fieldSeqComparators;
-        this.fieldAggregators = fieldAggregators;
+        this.fieldSeqComparators = getFieldSeqComparators(fieldSeqComparators);
+        this.fieldAggregators = getFieldAggregators(fieldAggregators);
         this.fieldSequenceEnabled = fieldSequenceEnabled;
         this.removeRecordOnDelete = removeRecordOnDelete;
         this.sequenceGroupPartialDelete = sequenceGroupPartialDelete;
         this.nullables = nullables;
     }
 
+    private List<WrapperWithFieldIndex<FieldsComparator>> getFieldSeqComparators(
+            Map<Integer, FieldsComparator> fieldSeqComparators) {
+        List<WrapperWithFieldIndex<FieldsComparator>> res = new ArrayList<>();
+        fieldSeqComparators.forEach(
+                (index, comparator) -> {
+                    res.add(new WrapperWithFieldIndex<>(comparator, index));
+                });
+        Collections.sort(res);
+        return res;
+    }
+
+    private List<WrapperWithFieldIndex<FieldAggregator>> getFieldAggregators(
+            Map<Integer, FieldAggregator> fieldAggregators) {
```

Comment:
getFieldSeqComparators and getFieldAggregators can be merged into one method using generics.